src/server/status_server.rs

```diff
@@ -59,19 +64,88 @@ impl StatusServer {
         let builder = Server::try_bind(&addr)?;
 
         // Create a status service.
-        let service = |req: Request<Body>| -> FutureResult<Response<Body>, hyper::Error> {
-            let mut response = Response::new(Body::empty());
+        let service = |req: Request<Body>| -> Box<Future<Item=Response<Body>, Error=hyper::Error> + Send> {
```

Comment:
Maybe use `impl Future<..>`?

Reply:
@breeswish So I can simply switch the Box to it? Do you have some resources online to learn about using `impl Future<..>` and why it could be better than Box? My knowledge on traits is very limited :)

Reply:
It seems when I remove the Box and use impl Future, I got error like this:
```
error[E0308]: match arms have incompatible types
  --> src/main.rs:25:16
   |
25 |           return match req.method() {
   |  ________________^
26 | |             &Method::PUT => {
27 | |                 req.into_body().concat2().map(move |chunk| {
28 | |                     let (_, name) = uri.split_at(FAIL_POINTS_PATH.len());
...  |
84 | |             }
85 | |         }
   | |_________^ expected struct `futures::Map`, found struct `futures::FutureResult`
   |
   = note: expected type `futures::Map<futures::stream::Concat2<hyper::Body>, [closure@src/main.rs:27:47: 53:18 uri:_]>`
              found type `futures::FutureResult<hyper::Response<_>, _>`
```
Basically I have [this piece of code](https://github.com/liufuyang/fail-rs-http-demo) to try out changes, made by example code [here](https://github.com/hyperium/hyper/blob/master/examples/params.rs)

Reply:
I tried to use `impl Future` on my small example code base:
https://github.com/liufuyang/fail-rs-http-demo/blob/try-use-impl-future/src/main.rs

You can clone it and compile it to see the error.

The problem seems to be `req.into_body().concat2().map(...)` returns some `futures::Map` type that is different with `futures::FutureResult`. I am not sure how to turn this Map into a FutureResult type. 

The code as it is now uses a Box with dyn Future and it seems working.